### PR TITLE
GPP serving TCF strings

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://ethyca.atlassian.net/browse/PROD-1285

### Description Of Changes

https://www.loom.com/share/ce626a97f570422389e9b0116a3f5bde

### Code Changes

* [x] Initialize the GPP CMP object
* [x] Subscribe to Fides events in the GPP extension
* [x] Follow example spec flow from https://github.com/InteractiveAdvertisingBureau/Global-Privacy-Platform/blob/main/Core/CMP%20API%20Specification.md
* [ ] Tests: added a follow up ticket since it's hard to write Cypress tests for this when it's not integrated yet https://ethyca.atlassian.net/browse/PROD-1420

### Steps to Confirm

* [ ] Watch the loom video—there are a few small changes in the code needed to get this working. This feature isn't "enabled" yet, and can only be done manually this way.
* [ ] Try out the various gpp functions
* [ ] Opt in/out of consents and check the gpp string. you can decode it here https://iabgpp.com/#

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Issue Requirements are Met
* [x] Update `CHANGELOG.md`

### CODE REVIEW

File: clients/fides-js/src/extensions/gpp.ts [hunk 1]
```diff
@@ -1,16 +1,56 @@
...
+import {
+  CmpApi,
+  CmpDisplayStatus,
+  CmpStatus,
+  SignalStatus,
+} from "@iabgpp/cmpapi";
 import { makeStub } from "../lib/gpp/stub";
+import { fidesEventToTcString } from "../lib/tcf/events";
+
+const CMP_ID = 407; // TODO: is this supposed to be the same as TCF, or is this separate?
```
Comment (allisonking) [inline clients/fides-js/src/extensions/gpp.ts:18]: Rachel will confirm this on Monday! https://ethyca.slack.com/archives/C05E74TR0BB/p1700245618221449

File: clients/fides-js/src/lib/tcf.ts [hunk 2]
```diff
@@ -149,25 +149,6 @@ export const generateFidesString = async ({
   return Promise.resolve(encodedString);
 };
 
-/**
```
Comment (allisonking) [inline clients/fides-js/src/lib/tcf.ts:152]: moved to a separate file so gpp can also use this function

File: clients/fides-js/src/components/Overlay.tsx [hunk 3]
```diff
@@ -49,15 +49,18 @@ const Overlay: FunctionComponent<Props> = ({
   const hasMounted = useHasMounted();
   const [bannerIsOpen, setBannerIsOpen] = useState(false);
 
-  const dispatchCloseEvent = useCallback(() => {
-    dispatchFidesEvent("FidesModalClosed", cookie, options.debug);
-  }, [cookie, options.debug]);
+  const dispatchCloseEvent = useCallback(
+    ({ saved = false }: { saved?: boolean }) => {
+      dispatchFidesEvent("FidesModalClosed", cookie, options.debug, { saved });
```
Comment (allisonking) [inline clients/fides-js/src/components/Overlay.tsx:54]: we need to know if the modal was closed without saving. gpp signals that it is not ready when the modal opens. once it saves, it signals it is ready again. but if you close the modal without saving, we'd be stuck in a "not ready" state

File: clients/fides-js/src/extensions/gpp.ts (hunk 1 above)
Comment (eastandwestwind) [reply clients/fides-js/src/extensions/gpp.ts:22]: she has confirmed! We can just use our TCF CMP ID, so let's remove this comment

File: clients/fides-js/src/lib/initialize.ts [hunk 4]
```diff
@@ -368,7 +368,7 @@ export const initialize = async ({
     identity: cookie.identity,
     fides_string: cookie.fides_string,
     tcf_consent: cookie.tcf_consent,
-    experience,
+    experience: effectiveExperience,
```
[the remark this answers is not in the document]
Comment (eastandwestwind) [reply clients/fides-js/src/lib/initialize.ts:371]: This looks good. The return obj built here is only used to update the `window.Fides` obj, which we don't rely on to make any functional decisions within or outside of Fides. Regardless, we should absolutely return the most up-to-date experience here. Thanks!

**Review by eastandwestwind — APPROVED** [review]
Great work on this @allisonking ! 

Do we have a separate ticket to add e2e tests, once we actually start making calls to the GPP lib?

Thumbs up from me, just need to remove an unneeded code comment 👍

**Comment by allisonking** [comment]
Thanks for the review @eastandwestwind ! I'll clean that comment up.

> Do we have a separate ticket to add e2e tests, once we actually start making calls to the GPP lib?

Yes! It is here: https://ethyca.atlassian.net/browse/PROD-1420

**Pull request merged**